mobile-academy/src/main/java/org/motechproject/nms/mobileacademy/service/impl/SmsNotificationServiceImpl.java

```diff
@@ -0,0 +1,162 @@
+
+        LOGGER.debug("Handling course completion notification event");
+        Long callingNumber = (Long) event.getParameters().get("callingNumber");
+        CompletionRecord cr = completionRecordDataService.findRecordByCallingNumber(callingNumber);
+
+        if (cr == null) {
+            // this should never be possible since the event dispatcher upstream adds the record
+            LOGGER.error("No completion record found for callingNumber: " + callingNumber);
+        }
+
+        cr.setSentNotification(sendNotificationRequest(callingNumber));
+        completionRecordDataService.update(cr);
+    }
+
+    @MotechListener(subjects = { SMS_STATUS })
+    public void updateSmsStatus(MotechEvent event) {
+        LOGGER.debug("Handling update sms delivery status event");
+
+        String callingNumber = (String) event.getParameters().get("address");
+        int startIndex = callingNumber.indexOf(':') + 2;
+        callingNumber = callingNumber.substring(startIndex);
+        CompletionRecord cr = completionRecordDataService.findRecordByCallingNumber(
+                Long.parseLong(callingNumber));
+
+        cr.setLastDeliveryStatus((String) event.getParameters().get("deliveryStatus"));
+        completionRecordDataService.update(cr);
+    }
+
+    private boolean sendNotificationRequest(Long callingNumber) {
+
+        CloseableHttpClient httpClient = HttpClients.createDefault();
+        HttpPost httpPost = prepareSmsRequest(callingNumber);
+
+        if (httpPost == null) {
+            LOGGER.error("Unable to build POST request for SMS notification");
+            return false;
+        }
+
+        try {
+            HttpResponse response = httpClient.execute(httpPost);
+            int responseCode = response.getStatusLine().getStatusCode();
+            if (responseCode != HttpStatus.SC_ACCEPTED) {
+                String error = String.format("Expecting HTTP 201 response from %s but received HTTP %d : %s ",
+                        httpPost.getURI().toString(), responseCode, EntityUtils.toString(response.getEntity()));
+                LOGGER.error(error);
+                if (response.getEntity() != null && response.getEntity().getContentLength() > 0) {
+                    // TODO: raise an alert
+                    LOGGER.error(getStringFromStream(response.getEntity().getContent()));
+                }
+
+                return false;
+            }
+        } catch (IOException ie) {
+            LOGGER.error(ie.toString());
+            return false;
+        }
+
+        return true;
+    }
+
+    private HttpPost prepareSmsRequest(Long callingNumber) {
+
+        String senderId = settingsFacade.getProperty(SMS_SENDER_ID);
+        String endpoint = settingsFacade.getProperty(SMS_NOTIFICATION_URL).replace("senderId", senderId);
+        String messageContent = settingsFacade.getProperty(SMS_MESSAGE_CONTENT);
+        String callbackEndpoint = settingsFacade.getProperty(CALLBACK_URL);
```

Comment:
Shouldn't you validate that the settings exist and are valid and alarm if they aren't?